### ISSUE TITLE
test(web): feature-branch regression testing - ES modules (🧩), main suite

### ISSUE THREAD

**Issue by jahorton** [keymanapp/keyman#9037]
# Keyman for Web Acceptance Test Procedures #

For this test run, please refer to the builds posted on https://github.com/keymanapp/keyman/pull/8560#issuecomment-1492750092.

As the feature branch has relocated test page resources, note that the links contained herein are updated to point to their new locations.  A few extra tests have also been added.

Once the feature-branch lands, updating the regression test spec to use the links from this one will resolve #9023 moving forward.  Some of the links may not work on `master`, though, hence my hesitancy to update them at the moment.

I have dropped SUITE_INSPECT for this run, as it's not against an official release of Keyman, but the rest of these tests should otherwise match what is seen at https://github.com/keymanapp/keyman/wiki/Platform-Acceptance-Test-for-Web at this time.

# User Testing

## SUITE_DESKTOP: Desktop Platforms

Ensure that all tests in the selections below are run on a representative set of the following browsers and setups:

* GROUP_WINDOWS_FIREFOX
* GROUP_WINDOWS_CHROME
* GROUP_MACOS_SAFARI
* GROUP_MACOS_CHROME
* GROUP_MACOS_FIREFOX
* GROUP_LINUX_FIREFOX

### Baseline Tests 

From the testing index page, use the first link ("View Keymanweb use samples") and then select "[Example 1](https://build.palaso.org/repository/download/Keymanweb_TestPullRequests/390024:id/src/samples/simplest/)" for the following tests.

* **TEST_BASELINE_ADD_SIL_IPA:** Using the "Add a keyboard by keyboard name" section, add `sil_ipa`.
  - Ensure the key sequence <kbd>n</kbd><kbd>></kbd> produces `ŋ`.

* **TEST_BASELINE_ADD_KM:** Using the "Add a keyboard by BCP-47 language code" section, add `km`. A Khmer keyboard should be added (perhaps khmer_basic, but not necessarily).

* **TEST_BASELINE_ADD_KHMER_ANGKOR:** Using the "Add a keyboard by keyboard name" section, add `khmer_angkor`.
  - Ensure the key combination <kbd>Ctrl</kbd>+<kbd>Alt</kbd><kbd>B</kbd> produces `ឞ`. Delete the character again.
  - Type the following (US) keys in sequence: <kbd>s</kbd> <kbd>u</kbd> <kbd>j</kbd> <kbd>r</kbd> <kbd>k</kbd>. You should see `ស្រុក`.
  - Continuing from the last test, hit <kbd>Backspace</kbd>, checking each time to ensure the output is as expected.  As this is a reorder (keyboard) rule test, you should see:
    - `ស្រុ`
    - `ស្រ`
    - `ស្`
    - `ស`
    - empty text area.

* **TEST_BASELINE_ADD_SPANISH:** Using the "Add a keyboard by language name(s) section", add `spanish`. Verify that a Spanish keyboard is added to the list.

* **TEST_BASELINE_SWEDISH:** Swap to the Swedish keyboard.
  - Ensure the key sequence <kbd>'</kbd> <kbd>e</kbd> produces `é` and that <kbd>o</kbd> <kbd>\\</kbd> <kbd>e</kbd> produces `œ`.

### Text Selection Tests

These tests are all run with the **web_context_tests** keyboard on the [Test unminified Keymanweb test page](https://build.palaso.org/repository/download/Keymanweb_TestPullRequests/390024:id/src/test/manual/web/unminified.html). They should be run on both INPUT and TEXTAREA elements. **Press Ctrl+F5 to reload the page between tests**.

* **TEST_SELECTION:** Type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd>x</kbd>. Select `bc` and type <kbd>q</kbd>. The output after the final key should be `aqx`. (This is testing selection replacement with no matching keyboard rule.)

* **TEST_SELECTION_2:** Type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd>x</kbd>. Select `bc` and type <kbd>p</kbd>. The output after the final key should be `aqx`. (This is testing selection replacement with a matching keyboard rule.)

* **TEST_CONTEXT_BASE:** Type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd>d</kbd>. The output after the final key should be `!`.

* **TEST_CONTEXT_SELECTION:** Type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd>x</kbd>. Select the `x` character, and type <kbd>d</kbd>. The output after the final key should be `abcd`.

* **TEST_CONTEXT_SELECTION_2:** Type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd>x</kbd>. Select the `x` character, and delete it with <kbd>Backspace</kbd>. Type <kbd>d</kbd>. The output after the final key should be `!`.

* **TEST_CONTEXT_SELECTION_3:** Type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd>x</kbd>. Select the `x` character, and type <kbd>y</kbd>. Press <kbd>Backspace</kbd>, and type <kbd>d</kbd>. The output after the final key should be `!`.

* **TEST_CONTEXT_SELECTION_4:** Type <kbd>x</kbd><kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd>x</kbd>. Select the `abc` characters, and type <kbd>d</kbd>. The output after the final key should be `xdx`.

* **TEST_SELECTION_SHIFT:** (**_new_**) Type <kbd>a</kbd><kbd>b</kbd><kbd>c</kbd><kbd>x</kbd>.  Select `bc` and then press the left <kbd>Shift</kbd> key on the OSK.  The output after pressing shift should be `abcx`, with `bc` still selected.

* **TEST_TYPING_AUTOSCROLL:** (**_new_**) Type a very long string of text into the "input field" element using the "English - English" keyboard until the text reaches its right-hand side.  
    - Type three additional letters.  
    - Verify that the text is automatically scrolled so that the last three characters and the caret (the thin, blinking `|` mark for text position) are visible.

### Element Attachment Tests

* **TEST_ATTACHMENT_API:** Utilize the [Attachment/Enablement API functionality](https://build.palaso.org/repository/download/Keymanweb_TestPullRequests/390024:id/src/test/manual/web/attachment-api/index.html?mode=auto) samples page to test the following:
  - `'auto'` attachment mode properly and automatically attaches to each dynamically-added control.
  - `'manual'` attachment mode results in a page with no KMW-activated controls.
  - Each 'attached' control functions as expected for the keyboard(s) selected.
  - KeymanWeb properly attaches to and detaches from each relevant type of control with the page's API-interface controls.
  - Confirm that the independent-keyboard management API functions correctly.

### Site-based tests of attachment API

All behavior exhibited here should feel "normal".  These tests are solely to ensure that nothing got accidentally broken.

For each of these, make use of the [Attachment/Enablement API functionality](https://build.palaso.org/repository/download/Keymanweb_TestPullRequests/390024:id/src/test/manual/web/attachment-api/index.html?mode=auto) test page.

- **TEST_NORMAL_USE**:  Ensure that the keyboard displays and works correctly when first loaded up.

    1.  Click the "Create Inputs" button one time.
    2. Click / touch the new page element.  The OSK should display.
    3. Use the OSK to type one or two keys.  The keys should produce their expected output.
    4. Click / touch a blank area of the page.  The OSK should automatically hide.

- **TEST_ELEMENT_HOPPING**:

    1.  Click the "Create Inputs" button one time and the "Create Textarea" button one time.
    2. Click / touch the first of the new page elements.  The OSK should display.
    3. Click / touch the other page element.  The OSK should remain visible.
    4. Click / tough the first element again.  The OSK should remain visible.
    5. Click / touch a blank area of the page.  The OSK should automatically hide.

- **TEST_SPECIFIC_KEYBOARDS**:

    1.  Click the "Create Inputs" button three times.
    2. Click / touch `Dynamic area #0.`
    3. Click the element's "Set to Dzongkha" button.
    4. Change the keyboard to French.
    5. Click `Dynamic area #1`'s "Set to Dzongkha button.
    6. Click `Dynamic area #1`.  The Dzongkha keyboard should be displayed.
    7. Click `Dynamic area #2`.
    8. Change the keyboard to Lao.
    9. Click `Dynamic area #0.`  The French keyboard should be displayed.
    10. Click `Dynamic area #2.`  The Lao keyboard should be displayed.
    11. Click `Dynamic area #1.`  The Dzongkha keyboard should be displayed.
    12. Click `Dynamic area #0`'s "Clear Keyboard" button.
    13. Click `Dynamic area #0.`  The Lao keyboard should be displayed.
  
### Tests CJK keyboard functionality

Use the standard [Test unminified Keymanweb test page](https://build.palaso.org/repository/download/Keymanweb_TestPullRequests/390024:id/src/test/manual/web/unminified.html).

- **TEST_JAPANESE_TYPING**:  Verify that a CJK keyboard still works.

  1. Under "Add a keyboard by keyboard name", type `japanese`, then click Add.
  2. Select the Japanese keyboard.  The OSK should change shape notably:
  
    <img width="347" alt="image" src="https://user-images.githubusercontent.com/25213402/131969825-d6705f1a-abef-4059-842f-e387e57ae67b.png">
    
  3. Type `a`.  A "picker" displaying a few options should display:

    <img width="460" alt="image" src="https://user-images.githubusercontent.com/25213402/131969902-1be50127-d09d-40be-9fce-b3b6b8840cf0.png">
    
  4. Type `2`.  The second option should replace the context.

- **TEST_JAPANESE_FOCUS**:  Verify that a CJK keyboard is shown and hidden properly without interfering with a page's UX.

  1. Select the page's textarea element.  The OSK's title bar should appear (as with step 2 for the `_TYPING` test).
  2. Click a blank area on the page.  The OSK should automatically hide and the caret should disappear from the textarea.
    -  Just in case, "caret":  the blinking vertical line segment that lets you know where new characters will appear in the context.
  3. Select the page's input field (the second editable control).  The OSK should reappear under the newly-selected control.
  4. Select the page's textarea element again.  The OSK should remain visible, and the caret should move from the input control to the textarea control.

## SUITE_TOUCH: Touch platform tests

Ensure that all tests in the selections below are run on a representative set of the following browsers and setups:

* GROUP_ANDROID_CHROME_TOUCH
* GROUP_ANDROID_CHROME_HARDWARE
* GROUP_IPHONE_SAFARI_TOUCH
* GROUP_IPHONE_SAFARI_HARDWARE
* GROUP_IPAD_SAFARI_TOUCH
* GROUP_IPAD_SAFARI_HARDWARE

### Baseline Tests 

From the testing index page, use the first link ("View Keymanweb use samples") and then select "[Example 1](https://build.palaso.org/repository/download/Keymanweb_TestPullRequests/390024:id/src/samples/simplest/)" for the following tests.

* **TEST_BASELINE_ADD_SIL_IPA:** Using the "Add a keyboard by keyboard name" section, add `sil_ipa`.
  - For physical devices, ensure the key sequence <kbd>n</kbd><kbd>></kbd> produces `ŋ`.
  - For touch devices, <kbd>ŋ</kbd> should be a subkey of <kbd>n</kbd> - ensure it works.

* **TEST_BASELINE_ADD_KM:** Using the "Add a keyboard by BCP-47 language code" section, add `km`. A Khmer keyboard should be added (perhaps khmer_basic, but not necessarily).

* **TEST_BASELINE_ADD_KHMER_ANGKOR:** Using the "Add a keyboard by keyboard name" section, add `khmer_angkor`.
  - If using a physical keyboard, ensure the key combination <kbd>Ctrl</kbd>+<kbd>Alt</kbd><kbd>B</kbd> produces `ឞ`. Delete the character again.
  - If the keyboard has four rows (phone form-factor):  Type in the following sequence on touch:  <kbd>ស</kbd> <kbd> ុ</kbd> <kbd> ្រ</kbd> (subkey of <kbd>រ</kbd>) <kbd>ក</kbd>.  You should see `ស្រុក`.
  - If the keyboard has five rows (tablet form-factor): <kbd>s</kbd> <kbd>u</kbd> <kbd>j</kbd> <kbd>r</kbd> <kbd>k</kbd>. You should see `ស្រុក`.
      - If the number of rows shown does not match the type of device you're using, fail the test for this.
  - Continuing from the last test, hit <kbd>Backspace</kbd>, checking each time to ensure the output is as expected.  As this is a reorder (keyboard) rule test, you should see:
    - `ស្រុ`
    - `ស្រ`
    - `ស`
    - empty text area.

* **TEST_BASELINE_ADD_SPANISH:** Using the "Add a keyboard by language name(s) section", add `spanish`. Verify that a Spanish keyboard is added to the list.

* **TEST_BASELINE_SWEDISH:** Swap to the Swedish keyboard.
  - If using a physical keyboard, ensure the key sequence <kbd>'</kbd> <kbd>e</kbd> produces `é` and that <kbd>o</kbd> <kbd>\\</kbd> <kbd>e</kbd> produces `œ`.
  - If using a touch-based keyboard, use long-press <kbd>.</kbd> to output <kbd>'</kbd> and then press <kbd>e</kbd>.  The two characters should not combine.
  - If using a touch-based keyboard, ensure that long-press <kbd>p</kbd> and long-press <kbd>G</kbd> displays properly and that the subkeys produce the expected output.

* **TEST_BASELINE_CAMEROON:** If using a touch-based keyboard, add the following keyboard by name:  "sil_cameroon_qwerty"
  - Ensure the top two rows of the symbol layer work correctly.  (These have unusual 'layer' settings.)
  - Ensure that the tri-color key (on both 'default' and 'shift' layers) changes layer appropriately and that keys pressed automatically transition back to the default layer.  (The 'default' layer's variant is very unusual, as it matches a keyboard rule and changes layer, but doesn't generate direct output!)
  - Test that a subkey on the 'default' and on the 'shift' layers output the expected character.

* **TEST_BASELINE_AUTOSCROLL:**  (**_new_**) Type a very long string of text into the "input field" element using the "English - English" keyboard until the text reaches its right-hand side.  
    - Type three additional letters.  
    - Verify that the text is automatically scrolled so that the last three characters and the caret (the thin, blinking `|` mark for text position) are visible.
    - Verify that the keyboard does not disappear at any point while typing and that it is still visible.

* **TEST_ROTATION:** (**_new_**)  Put the test device into 'portrait' orientation - with the longer dimension held vertically.
    * Type <key>a</key> into the input field.
    * Rotate your device to a landscape (horizontal) orientation.
    * Verify that the OSK repositions itself at the bottom of the screen and occupies the full width of the device.
    * Rotate the device back to a portrait (vertical) orientation.
    * Again, verify that the OSK repositions itself at the bottom of the screen and occupies the full width of the device.
        - After rotation, it is possible that the previously-typed 'a' will not be visible.  
        - If so, please create a separate issue for this, but do not fail the test.  (It appears to be a 16.0 issue, too.)

### Element Attachment Tests

* **TEST_ATTACHMENT_API:** Utilize the [Attachment/Enablement API functionality](https://build.palaso.org/repository/download/Keymanweb_TestPullRequests/390024:id/src/test/manual/web/attachment-api/index.html?mode=auto) test page to test the following:
  - `'auto'` attachment mode properly and automatically attaches to each dynamically-added control.
  - `'manual'` attachment mode results in a page with no KMW-activated controls.
  - Each 'attached' control functions as expected for the keyboard(s) selected.
  - KeymanWeb properly attaches to and detaches from each relevant type of control with the page's API-interface controls.
  - Confirm that the independent-keyboard management API functions correctly.

### Site-based tests of attachment API
All behavior exhibited here should feel "normal".  These tests are solely to ensure that nothing got accidentally broken.

For each of these, make use of the [Attachment/Enablement API functionality](https://build.palaso.org/repository/download/Keymanweb_TestPullRequests/390024:id/src/test/manual/web/attachment-api/index.html?mode=auto) test page.

- **TEST_NORMAL_USE**:  Ensure that the keyboard displays and works correctly when first loaded up.

  1.  Click the "Create Inputs" button one time.
  2. Click / touch the new page element.  The OSK should display.
  3. Use the OSK to type one or two keys.  The keys should produce their expected output.
  4. Click / touch a blank area of the page.  The OSK should automatically hide.

- **TEST_ELEMENT_HOPPING**:

  1.  Click the "Create Inputs" button one time and the "Create Textarea" button one time.
  2. Click / touch the first of the new page elements.  The OSK should display.
  3. Click / touch the other page element.  The OSK should remain visible.
  4. Click / tough the first element again.  The OSK should remain visible.
  5. Click / touch a blank area of the page.  The OSK should automatically hide.

- **TEST_SPECIFIC_KEYBOARDS**:

  1.  Click the "Create Inputs" button three times.
  2. Click / touch `Dynamic area #0.`
  3. Click the element's "Set to Dzongkha" button.
  4. Change the keyboard to French.
  5. Click `Dynamic area #1`'s "Set to Dzongkha button.
  6. Click `Dynamic area #1`.  The Dzongkha keyboard should be displayed.
  7. Click `Dynamic area #2`.
  8. Change the keyboard to Lao.
  9. Click `Dynamic area #0.`  The French keyboard should be displayed.
  10. Click `Dynamic area #2.`  The Lao keyboard should be displayed.
  11. Click `Dynamic area #1.`  The Dzongkha keyboard should be displayed.
  12. Click `Dynamic area #0`'s "Clear Keyboard" button.
  13. Click `Dynamic area #0.`  The Lao keyboard should be displayed.

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_WINDOWS_FIREFOX: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that the key sequence n> produced ŋ.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/523459ce-3a53-4f74-8b71-6d5aa1b98325">


* **TEST_BASELINE_ADD_KM (PASSED):** Verified that the Khmer_Basic keyboard was added.
  
   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/f8830dc9-b618-4f81-9ea6-2c9562122cc1">

* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that the Khmer_angkor keyboard was added. Able to see ស្រុក by typing key sequence s u j r k. Able  to see that the reorder rule test, by hitting Backspace key. 

* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that the Spanish keyboard was added to the list. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/8aae0e67-9a63-4e47-8bb2-7078676d6fdf">

  
* **TEST_BASELINE_SWEDISH (PASSED):** Verified that the given key sequences shows the expected output.

 <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/dabb1cfd-00c0-42c9-91f7-9f73a641af69">

  <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/aa68d328-dadd-4c98-a40a-1a9d6ec062db">

**Comment by bharanidharanj** [comment]
### GROUP_WINDOWS_CHROME: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that the key sequence n> produced ŋ.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/e348461c-e3a2-4dda-b585-acb9cdf03708">


* **TEST_BASELINE_ADD_KM (PASSED):** Verified that the Khmer_Basic keyboard was added.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/48aea182-5d9f-4007-ab5b-0ca75a67da7b">


* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that the Khmer_angkor keyboard was added. Able to see ស្រុក by typing key sequence s u j r k. Able  to see that the reorder rule test, by hitting Backspace key.

* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that the Spanish keyboard was added to the list.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/85926f3a-6ae1-4052-8431-bf60ef7c5c6a">


* **TEST_BASELINE_SWEDISH (PASSED):** Verified that the given key sequences shows the expected output.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/85f2e8b6-bcf2-4577-837c-5b7e3ce63180">

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/c536d0e4-c8d6-4384-98d2-1ffa15b12ef1">

**Comment by bharanidharanj** [comment]
### GROUP_LINUX_FIREFOX: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that the key sequence n> produced ŋ.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/6652ae81-9bb8-4e64-a5ae-28b970f20216">

* **TEST_BASELINE_ADD_KM (PASSED):** Verified that the Khmer_Basic keyboard was added. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/b263ff63-cfa6-4343-9b39-245089d0c3c2">


* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that the Khmer_angkor keyboard was added. Able to see ស្រុក by typing key sequence s u j r k. Able  to see that the reorder rule test, by hitting Backspace key. 

* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that the Spanish keyboard was added to the list. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/33dab923-4e63-4c63-91a2-e140b371b8bb">

* **TEST_BASELINE_SWEDISH (PASSED):** Verified that the given key sequences shows the expected output. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/9d7c3027-5f19-4970-813e-fda6586d2ceb">

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/d3086651-cb3a-4908-b049-8a3ac023d4dc">

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_MACOS_SAFARI: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that the key sequence n> produced ŋ.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/f01e4382-b61c-4768-8971-7839259d0481">


* **TEST_BASELINE_ADD_KM (PASSED):** Verified that the Khmer_Basic keyboard was added. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/20e8373a-6994-4b9e-b4f4-fc85cb3a609e">


* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that the Khmer_angkor keyboard was added. Able to see ស្រុក by typing key sequence s u j r k. Able  to see that the reorder rule test, by hitting Backspace key. 


* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that the Spanish keyboard was added to the list. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/1d222663-84c6-4f23-abbd-34aed034d488">



* **TEST_BASELINE_SWEDISH (PASSED):** Verified that the given key sequences shows the expected output. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/1647c8ad-0ba4-42a4-b1d2-cb08d5dad13b">

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/a81d6d87-3700-45df-87a3-d75fa0e64344">

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_MACOS_CHROME: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that the key sequence n> produced ŋ.
   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/07c19e3c-7fe5-4b3b-8e46-84c109f555c1">


* **TEST_BASELINE_ADD_KM (PASSED):** Verified that the Khmer_Basic keyboard was added.
   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/f661bb42-957f-4282-8e1b-949008224c20">


* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that the Khmer_angkor keyboard was added. Able to see ស្រុក by typing key sequence s u j r k. Able  to see that the reorder rule test, by hitting Backspace key.


* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that the Spanish keyboard was added to the list. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/c84ee29c-4ff8-45cc-bc72-1df5edf636fe">


* **TEST_BASELINE_SWEDISH (PASSED):** Verified that the given key sequences shows the expected output. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/c0efb2b4-bfd6-4905-8e42-ae2f0122c0f0">

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/6d953ce2-dd13-4639-902b-08e693147f10">


### GROUP_MACOS_FIREFOX: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that the key sequence n> produced ŋ.


* **TEST_BASELINE_ADD_KM (PASSED):** Verified that the Khmer_Basic keyboard was added.


* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that the Khmer_angkor keyboard was added. Able to see ស្រុក by typing key sequence s u j r k. Able  to see that the reorder rule test, by hitting Backspace key.


* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that the Spanish keyboard was added to the list. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/50c6c9b8-0c91-48c0-aae4-1b66dcf78720">

* **TEST_BASELINE_SWEDISH (PASSED):** Verified that the given key sequences shows the expected output. 
 
   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/ddeef2b3-5465-42f7-8a3a-201d50b004d6">

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/1fbac2f8-fd69-4fb9-baff-c5d33aa83577">

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_WINDOWS_FIREFOX: 

* **TEST_SELECTION (PASSED):** Tested in unminified Keymanweb test page and tested in both INPUT and TEXTAREA elements and here is my observations: 1. Verified that the output shows aqx.

* **TEST_SELECTION_2 (PASSED):** Verified that the output after the final key shows aqx.

* **TEST_CONTEXT_BASE (PASSED):** Verified that the output finally shows !. 

* **TEST_CONTEXT_SELECTION (PASSED):** Verified that the output shows abcd.

* **TEST_CONTEXT_SELECTION_2 (PASSED):** Verified that the final key shows !.

* **TEST_CONTEXT_SELECTION_3 (PASSED):** Verified that the final key shows !.

* **TEST_CONTEXT_SELECTION_4 (PASSED):** Verified that the final key shows xdx.

* **TEST_SELECTION_SHIFT (PASSED):** Verified that the pressing shift shows abcx with bc still selected.

![Shift_firefox](https://github.com/keymanapp/keyman/assets/19683143/baab165f-6c37-4549-adbc-22db25660071)


* **TEST_TYPING_AUTOSCROLL (PASSED):** Verified that the text autoscroll is working in the text area whereas the same is not working in INPUT field. However, the keyboard is still visible and working. 

..textarea

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/f7b0ab47-8f69-4303-b724-891c8661cd13">

..inputfield

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/5ac9751d-565e-4140-9e6f-1c15d24279a6">

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_WINDOWS_FIREFOX: 

* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly.

* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up. 

* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.

* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine. 

* **TEST_JAPANESE_TYPING (PASSED):** 1. Able to add japanese keyboard. 2. Able to select the second option to replace the existing context. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/cf8514b3-7a0c-424f-8728-c6b11aabfa67">

* **TEST_JAPANESE_FOCUS (PASSED):** 1. Verified that it is possible to see the japanese keyboard when I clicked either in text area or the input field. 2. If I clicked in the black area on the page, the keyboard disappeared. Seems to be working fine. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/ebf7b2a5-08f1-4595-9ccf-93e0617ac97d">

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_WINDOWS_CHROME: 

* **TEST_SELECTION (PASSED):** Tested in unminified Keymanweb test page and tested in both INPUT and TEXTAREA elements and here is my observations: 1. Verified that the output shows aqx.
* **TEST_SELECTION_2 (PASSED):** Verified that the output after the final key shows aqx. 
* **TEST_CONTEXT_BASE (PASSED):** Verified that the output finally shows !. 
* **TEST_CONTEXT_SELECTION (PASSED):** Verified that the output shows abcd.
* **TEST_CONTEXT_SELECTION_2 (PASSED):** Verified that the final key shows !.
* **TEST_CONTEXT_SELECTION_3 (PASSED):** Verified that the final key shows !.
* **TEST_CONTEXT_SELECTION_4 (PASSED):** Verified that the final key shows xdx.
* **TEST_SELECTION_SHIFT (PASSED):** Verified that the pressing shift shows abcx with bc still selected. 
* **TEST_TYPING_AUTOSCROLL (PASSED):** Verified that the text automatically scrolled in text area view and it does not scroll in input field.
* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly. 
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up. 
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area. 
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine. 
* **TEST_JAPANESE_TYPING (PASSED):** 1. Able to add japanese keyboard. 2. Able to select the second option to replace the existing context. 
* **TEST_JAPANESE_FOCUS (PASSED):** 1. Verified that it is possible to see the japanese keyboard when I clicked either in text area or the input field. 2. If I clicked in the black area on the page, the keyboard disappeared. Seems to be working fine.

**Comment by jahorton** [comment]
@bharanidharanj 

> * **TEST_TYPING_AUTOSCROLL (FAILED):** Verified that the text autoscroll is not working in INPUT field. 
>
> ..inputfield
>
>   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/5ac9751d-565e-4140-9e6f-1c15d24279a6">

... but it _did_ autoscroll.  Note the left-hand side of the input field.  I'm assuming that you just typed the word `testing` over and over again... so why is the first thing visible in the input field `ting`, not `testing`?  It's because the first copy's `tes` got "autoscrolled" off the left-hand side.

![image](https://github.com/keymanapp/keyman/assets/25213402/682cb4c0-5b32-47d9-8398-72357219914f)

Note how you can still see the caret at the end of the input field, too:

![image](https://github.com/keymanapp/keyman/assets/25213402/13c48781-4256-40fe-b89d-075fbb0ce289)

From the test's definition:

> Verify that the text is automatically scrolled so that the last three characters and the caret (the thin, blinking | mark for text position) are visible.

Your `ing` is visible, as well as the caret.  It totally PASSES.

**Comment by bharanidharanj** [comment]
> @bharanidharanj
> 
> > * **TEST_TYPING_AUTOSCROLL (FAILED):** Verified that the text autoscroll is not working in INPUT field.
> > 
> > ..inputfield
> >   <img alt="" width="300" src="https://user-images.githubusercontent.com/19683143/247508902-5ac9751d-565e-4140-9e6f-1c15d24279a6.PNG">
> 
> ... but it _did_ autoscroll. Note the left-hand side of the input field. I'm assuming that you just typed the word `testing` over and over again... so why is the first thing visible in the input field `ting`, not `testing`? It's because the first copy's `tes` got "autoscrolled" off the left-hand side.
> 
> ![image](https://user-images.githubusercontent.com/25213402/247832552-682cb4c0-5b32-47d9-8398-72357219914f.png)
> 
> Note how you can still see the caret at the end of the input field, too:
> 
> ![image](https://user-images.githubusercontent.com/25213402/247832687-13c48781-4256-40fe-b89d-075fbb0ce289.png)
> 
> From the test's definition:
> 
> > Verify that the text is automatically scrolled so that the last three characters and the caret (the thin, blinking | mark for text position) are visible.
> 
> Your `ing` is visible, as well as the caret. It totally PASSES.

@jahorton I thought after typing the last 3 characters the word 'testing' will jump into the next line (ie., move down to next line ..like in text area). Now, I can understand the procedure. Thanks for the Clarification. :-)

**Comment by jahorton** [comment]
@keymanapp-test-bot retest SUITE_DESKTOP TEST_TYPING_AUTOSCROLL

**Comment by bharanidharanj** [comment]
### GROUP_LINUX_FIREFOX: 

* **TEST_SELECTION (PASSED):** Tested in unminified Keymanweb test page and tested in both INPUT and TEXTAREA elements and here is my observations: 1. Verified that the output shows aqx.
* **TEST_SELECTION_2 (PASSED):** Verified that the output after the final key shows aqx. 
* **TEST_CONTEXT_BASE (PASSED):** Verified that the output finally shows !. 
* **TEST_CONTEXT_SELECTION (PASSED):** Verified that the output shows abcd. 
* **TEST_CONTEXT_SELECTION_2 (PASSED):** Verified that the final key shows !.
* **TEST_CONTEXT_SELECTION_3 (PASSED):** Verified that the final key shows !.
* **TEST_CONTEXT_SELECTION_4 (PASSED):** Verified that the final key shows xdx.
* **TEST_SELECTION_SHIFT (PASSED):** Verified that the pressing shift shows abcx with bc still selected.
* **TEST_TYPING_AUTOSCROLL (PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible.  2. Verified that the Keyboard is still visible at any point while typing in the input field. 
* **TEST_ATTACHMENT_API (PASSED):** API: 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly. 
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine. 
* **TEST_JAPANESE_TYPING (PASSED):** 1. Able to add japanese keyboard. 2. Able to select the second option to replace the existing context. 
* **TEST_JAPANESE_FOCUS (PASSED):** 1. Verified that it is possible to see the japanese keyboard when I clicked either in text area or the input field. 2. If I clicked in the black area on the page, the keyboard disappeared. Seems to be working fine.

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_WINDOWS_FIREFOX: 

* **TEST_TYPING_AUTOSCROLL (PASSED):** Retested as per Josh's suggestion and here my observation: 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible.  2. Verified that the Keyboard is still visible at any point while typing in the input field.

### GROUP_WINDOWS_CHROME: 

* **TEST_TYPING_AUTOSCROLL (PASSED):** Retested as per Josh's suggestion and here my observation: 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible.  2. Verified that the Keyboard is still visible at any point while typing in the input field.

### GROUP_MACOS_SAFARI: 

* **TEST_TYPING_AUTOSCROLL (PASSED):** Retested as per Josh's suggestion and here my observation: 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible.  2. Verified that the Keyboard is still visible at any point while typing in the input field.

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_MACOS_CHROME: 

* **TEST_SELECTION (PASSED):** Tested in unminified Keymanweb test page and tested in both INPUT and TEXTAREA elements and here is my observations: 1. Verified that the output shows aqx.
* **TEST_SELECTION_2 (PASSED):** Verified that the output after the final key shows aqx.
* **TEST_CONTEXT_BASE (PASSED):** Verified that the output finally shows !.
* **TEST_CONTEXT_SELECTION (PASSED):** Verified that the output shows abcd.
* **TEST_CONTEXT_SELECTION_2 (PASSED):** Verified that the final key shows !.
* **TEST_CONTEXT_SELECTION_3 (PASSED):** Verified that the final key shows !.
* **TEST_CONTEXT_SELECTION_4 (PASSED):** Verified that the final key shows xdx.
* **TEST_SELECTION_SHIFT (PASSED):** Verified that the pressing shift shows abcx with bc still selected.
* **TEST_TYPING_AUTOSCROLL (PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible.  2. Verified that the Keyboard is still visible at any point while typing in the input field.
* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly.
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine.
* **TEST_JAPANESE_TYPING (PASSED):** 1. Able to add japanese keyboard. 2. Able to select the second option to replace the existing context.
* **TEST_JAPANESE_FOCUS (PASSED):** 1. Verified that it is possible to see the japanese keyboard when I clicked either in text area or the input field. 2. If I clicked in the black area on the page, the keyboard disappeared. Seems to be working fine.

**Comment by bharanidharanj** [comment]
## SUITE_DESKTOP: Desktop Platforms

### GROUP_MACOS_FIREFOX: 

* **TEST_SELECTION (PASSED):** Tested in unminified Keymanweb test page and tested in both INPUT and TEXTAREA elements and here is my observations: 1. Verified that the output shows aqx.
* **TEST_SELECTION_2 (PASSED):** Verified that the output after the final key shows aqx.
* **TEST_CONTEXT_BASE (PASSED):** Verified that the output finally shows !.
* **TEST_CONTEXT_SELECTION (PASSED):** Verified that the output shows abcd.
* **TEST_CONTEXT_SELECTION_2 (PASSED):** Verified that the final key shows !.
* **TEST_CONTEXT_SELECTION_3 (PASSED):** Verified that the final key shows !.
* **TEST_CONTEXT_SELECTION_4 (PASSED):** Verified that the final key shows xdx.
* **TEST_SELECTION_SHIFT (PASSED):** Verified that the pressing shift shows abcx with bc still selected.
* **TEST_TYPING_AUTOSCROLL (PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible.  2. Verified that the Keyboard is still visible at any point while typing in the input field.
* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly.
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine.
* **TEST_JAPANESE_TYPING (PASSED):** 1. Able to add japanese keyboard. 2. Able to select the second option to replace the existing context.
* **TEST_JAPANESE_FOCUS (PASSED):** 1. Verified that it is possible to see the japanese keyboard when I clicked either in text area or the input field. 2. If I clicked in the black area on the page, the keyboard disappeared. Seems to be working fine.

**Comment by bharanidharanj** [comment]
## SUITE_TOUCH: Touch platform tests

### GROUP_ANDROID_CHROME_TOUCH: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that pressing the subkey of n - shows the expected output.
* **TEST_BASELINE_ADD_KM PASSED):** Verified that able to see the Khmer_basic keyboard.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/4f1f2e32-daf0-4078-b8b1-8a6b9b2753c0">

* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that typing the given sequence of letters on touch produces the expected output. Also, verified that pressing the Backspace key, delete the character one by one as per the reorder rule.
* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that a Spanish keyboard is added to the list.
* **TEST_BASELINE_SWEDISH (PASSED):** 1. Verified that the two characters does not combine. 2. Verified that long pressing p or long pressing G displays the subkey menu and clicking on that produces expected output.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/79717113-e99c-4434-9726-544a90c87bfb">

* **TEST_BASELINE_CAMEROON (PASSED):** 1. Verified that the top two rows of the symbol layer working correctly. 2. Verified that the tri-color key changes layer appropriately and that keys pressed automatically transition back to the default layer. 3. Verified that the subkey on the 'default' and on the 'shift' layers output the expected character.
* **TEST_BASELINE_AUTOSCROLL PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible. 2. Verified that the Keyboard is still visible at any point while typing in the input field.
* **TEST_ROTATION (PASSED):** 1. Verified that the OSK repositions itself at the bottom of the screen and occupies the full width of the device. (in both the roations -  horizontal and vertical mode)
* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly.
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/795dbcc5-b8d9-4637-bf53-c9199e884986">

**Comment by bharanidharanj** [comment]
## SUITE_TOUCH: Touch platform tests

### GROUP_ANDROID_CHROME_HARDWARE: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that pressing the subkey of n - shows the expected output.
* **TEST_BASELINE_ADD_KM (PASSED):** Verified that able to see the Khmer_basic keyboard.
* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that typing the given sequence of letters on touch produces the expected output. Also, verified that pressing the Backspace key, delete the character one by one as per the reorder rule.
* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that a Spanish keyboard is added to the list.
* **TEST_BASELINE_SWEDISH (PASSED):** 1. Verified that the two characters does not combine. 2. Verified that long pressing p or long pressing G displays the subkey menu and clicking on that produces expected output.
* **TEST_BASELINE_CAMEROON (PASSED):** 1. Verified that the top two rows of the symbol layer working correctly. 2. Verified that the tri-color key changes layer appropriately and that keys pressed automatically transition back to the default layer. 3. Verified that the subkey on the 'default' and on the 'shift' layers output the expected character.
* **TEST_BASELINE_AUTOSCROLL (PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible. 2. Verified that the Keyboard is still visible at any point while typing in the input field.
* **TEST_ROTATION (PASSED):** 1. Verified that the OSK repositions itself at the bottom of the screen and occupies the full width of the device. (in both the rotations -  horizontal and vertical mode)
* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly. 
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine.

**Comment by bharanidharanj** [comment]
## SUITE_TOUCH: Touch platform tests

### GROUP_IPHONE_SAFARI_TOUCH: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that pressing the subkey of n - shows the expected output.
* **TEST_BASELINE_ADD_KM (PASSED):** Verified that able to see the Khmer_basic keyboard.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/37d3453c-305e-475d-b08b-0eec1f42045d">

* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that typing the given sequence of letters on touch produces the expected output. Also, verified that pressing the Backspace key, delete the character one by one as per the reorder rule.
* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that a Spanish keyboard is added to the list.
* **TEST_BASELINE_SWEDISH (PASSED):** 1. Verified that the two characters does not combine. 2. Verified that long pressing p or long pressing G displays the subkey menu and clicking on that produces expected output.
* **TEST_BASELINE_CAMEROON (PASSED):** 1. Verified that the top two rows of the symbol layer working correctly. 2. Verified that the tri-color key changes layer appropriately and that keys pressed automatically transition back to the default layer. 3. Verified that the subkey on the 'default' and on the 'shift' layers output the expected character.
* **TEST_BASELINE_AUTOSCROLL (PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible. 2. Verified that the Keyboard is still visible at any point while typing in the input field.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/af5653b0-715e-4d38-8632-8e95b88c9534">

* **TEST_ROTATION (PASSED):** 1. Verified that the OSK repositions itself at the bottom of the screen and occupies the full width of the device. (in both the rotations - horizontal and vertical mode)

* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly.
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/568844ff-c82a-445a-bb2f-6c4fa11c63e4">

**Comment by bharanidharanj** [comment]
## SUITE_TOUCH: Touch platform tests

### GROUP_IPHONE_SAFARI_HARDWARE: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that pressing the subkey of n - shows the expected output.
* **TEST_BASELINE_ADD_KM (PASSED):** Verified that able to see the Khmer_basic keyboard.
* **TEST_BASELINE_ADD_KHMER_ANGKOR (PASSED):** Verified that typing the given sequence of letters on touch produces the expected output. Also, verified that pressing the Backspace key, delete the character one by one as per the reorder rule.
* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that a Spanish keyboard is added to the list.
* **TEST_BASELINE_SWEDISH (PASSED):** 1. Verified that the two characters does not combine. 2. Verified that long pressing p or long pressing G displays the subkey menu and clicking on that produces expected output.
* **TEST_BASELINE_CAMEROON (PASSED):** 1. Verified that the top two rows of the symbol layer working correctly. 2. Verified that the tri-color key changes layer appropriately and that keys pressed automatically transition back to the default layer. 3. Verified that the subkey on the 'default' and on the 'shift' layers output the expected character.
* **TEST_BASELINE_AUTOSCROLL (PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible. 2. Verified that the Keyboard is still visible at any point while typing in the input field.
* **TEST_ROTATION (PASSED):** 1. Verified that the OSK repositions itself at the bottom of the screen and occupies the full width of the device. (in both the rotations - horizontal and vertical mode)
* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly.
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine.

**Comment by bharanidharanj** [comment]
## SUITE_TOUCH: Touch platform tests

### GROUP_IPAD_SAFARI_TOUCH: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that pressing the subkey of n - shows the expected output.
* **TEST_BASELINE_ADD_KM (PASSED):** Verified that able to see the Khmer_basic keyboard.
* **TEST_BASELINE_ADD_KHMER_ANGKOR (FAILED):** Khmer Angkor keyboard does not show up the subkey. 

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/2925d713-0e3b-4580-b493-2339600c6910">

* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that a Spanish keyboard is added to the list.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/83fb817f-a507-4567-8559-147abd96e29b">

* **TEST_BASELINE_SWEDISH (PASSED):** 1. Verified that the two characters does not combine. 2. Verified that long pressing p or long pressing G displays the subkey menu and clicking on that produces expected output.
   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/e19c78c7-3d26-47a5-895e-318a0632a4a3">


* **TEST_BASELINE_CAMEROON (PASSED):** 1. Verified that the top two rows of the symbol layer working correctly. 2. Verified that the tri-color key changes layer appropriately and that keys pressed automatically transition back to the default layer. 3. Verified that the subkey on the 'default' and on the 'shift' layers output the expected character.
* **TEST_BASELINE_AUTOSCROLL (PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible. 2. Verified that the Keyboard is still visible at any point while typing in the input field.

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/2ab46fb4-ae78-4f15-943b-d418c8b35c44">


* **TEST_ROTATION (PASSED):** 1. Verified that the OSK repositions itself at the bottom of the screen and occupies the full width of the device. (in both the roations -  horizontal and vertical mode)
..Portrait
   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/08cd2fc2-acdc-4e8c-82d5-d84e90cb6550">

..Landscape

   <img width="300" src="https://github.com/keymanapp/keyman/assets/19683143/5ad57984-16a7-492c-b7e1-455144dad45c">


* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly. 
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine.
![API](https://github.com/keymanapp/keyman/assets/19683143/54cbb67b-a2c6-4e5d-bdf7-ef47e10bb1cc)

### GROUP_IPAD_SAFARI_HARDWARE: 

* **TEST_BASELINE_ADD_SIL_IPA (PASSED):** Verified that pressing the subkey of n - shows the expected output.
* **TEST_BASELINE_ADD_KM (PASSED):** Verified that able to see the Khmer_basic keyboard.
* **TEST_BASELINE_ADD_KHMER_ANGKOR (FAILED):** Khmer Angkor keyboard does not show up the subkey.
* **TEST_BASELINE_ADD_SPANISH (PASSED):** Verified that a Spanish keyboard is added to the list.
* **TEST_BASELINE_SWEDISH (PASSED):** 1. Verified that the two characters does not combine. 2. Verified that long pressing p or long pressing G displays the subkey menu and clicking on that produces expected output.
* **TEST_BASELINE_CAMEROON (PASSED):** 1. Verified that the top two rows of the symbol layer working correctly. 2. Verified that the tri-color key changes layer appropriately and that keys pressed automatically transition back to the default layer. 3. Verified that the subkey on the 'default' and on the 'shift' layers output the expected character.
* **TEST_BASELINE_AUTOSCROLL (PASSED):** 1. Verified that the text is automatically scrolled so that the last three characters and the caret are fully visible. 2. Verified that the Keyboard is still visible at any point while typing in the input field.
* **TEST_ROTATION (PASSED):** 1. Verified that the OSK repositions itself at the bottom of the screen and occupies the full width of the device. (in both the roations -  horizontal and vertical mode)
* **TEST_ATTACHMENT_API (PASSED):** 1. Verified that the 'auto' attachment mode properly and automatically attached to each dynamically-added control. 2. Verified the 'manual' attachment mode results in a page with no KMW-activated controls. 3. Verified that each 'attached' control functions as expected for the selected keyboard(s). 4. Verified that the KeymanWeb properly attaches to and detaches from each relevant type of control. Also, verified that the independent-keyboard management API functions correctly.
* **TEST_NORMAL_USE (PASSED):** Tested in the given test page and here is my observation : 1. Verified that the keyboard displays and works correctly when the first loaded up.
* **TEST_ELEMENT_HOPPING (PASSED):** Verified that the OSK remains visible if I jump to another page element and it is automatically hidden if I clicked on the blank area.
* **TEST_SPECIFIC_KEYBOARDS (PASSED):** Verified that the keyboard changes from one dynamic area to other and it is showing the expected keyboard layout. Seems to be working fine.

**Comment by jahorton** [comment]
OK... so the two test failures are because the tablet form-factor layout for `khmer_angkor` doesn't use subkeys to the same extent as the phone form-factor layout.  I've gone ahead and adjusted the description accordingly.

Looks like everything else passed swimmingly, so... I think that means we're good to go.

**Comment by jahorton** [comment]
For updating the [Platform Acceptance Test for Web wiki page](https://github.com/keymanapp/keyman/wiki/Platform-Acceptance-Test-for-Web):
    - Do include the wiki page's `SUITE_INSPECT` in its original position.
    - Replace all other SUITEs with their specs from the description of this PR.
    - Do a find-and-replace on all Web test-page links when copying the description text over:  `Keymanweb_TestPullRequests/390024:id/src` => `Keymanweb_TestPullRequests/TBD:id/src`, since that's the pattern we've been using for the platform-acceptance user testing template.

**Comment by jahorton** [comment]
#8560 has merged!  🎊 

As this was a testing issue for that branch, we can retire this issue now.
